src/proxy.c

```diff
@@ -47,6 +47,7 @@ struct watcher_proxy_data
 	 * otherwise it WILL wait
 	 */
 	int         *oob_fd;
+	gboolean    oob_fd_found;
```

Comment:
We do 2 reads for the 2 separate messages that the proxy send us on its blocking socket. So at first we read the OOB ancillary data and the fd, and then get the 2nd message for reading the actual message. I changed the code slightly to make that clearer, with even code comments :)